src/common/extraction/extractNxGraph.ts

```diff
@@ -0,0 +1,70 @@
+import * as fs from "fs"
+import * as path from "path"
+import { Graph } from "./graph"
+
+type Nodes = {
+	[index: string]: [Outgoing]
```

Comment:
What's the point of using singleton arrays here? Why not `[index: string]: Outgoing`?

In other words, why is this type `Record<string, [Outgoing]>` and not `Record<string, Outgoing>`?

Reply:
That was a bad habit from Haskell, it should be `Outgoing[]`.